spot_driver/spot_driver/spot_ros2.py

```diff
@@ -2521,6 +2525,30 @@ def body_pose_callback(self, data: Pose) -> None:
         mobility_params.body_control.CopyFrom(body_control)
         self.spot_wrapper.set_mobility_params(mobility_params)
 
+    def arm_joint_cmd_callback(self, data: JointState) -> None:
+        if not self.spot_wrapper:
+            self.get_logger().info(f"Mock mode, received arm joint commdn {data}")
+            return
+        arm_joint_map = {"sh0": 0.0, "sh1": 0.0, "el0": 0.0, "el1": 0.0, "wr0": 0.0, "wr1": 0.0}
+        if len(data.name) != len(arm_joint_map):
+            self.get_logger().warning(f"Expected {len(arm_joint_map)} joints, but received {len(data.name)}")
+            return
+
+        for joint in zip(data.name, data.position):
+            for joint_name in arm_joint_map.keys():
+                if joint_name in joint[0]:
+                    arm_joint_map[joint_name] = joint[1]
+                    continue
```

Comment:
Thank you! I knew there had to be a better way to do this

Reply:
@amessing-bdai I tried your proposed change but I remember why I had to do it the way I did :( Because of namespacing and slight differences in joint naming conventions - some spot models use `arm_sh0` while others use just `sh0` so the only consistent thing is the shx, elx, wrx without knowing for sure if it's precursored with the `arm_` it's hard to match it in the way you said. Also because names come in as an array of strings from the ros standard JointState msg, it is often precursored with a namespace as well

Reply:
Ok. I still would recommend error checking that each joint was set. Maybe you initialize them all to None and just check that they were all set before sending them to spot_wrapper?

Reply:
Also can you add a comment that explains that? Will potentially help the next person who looks at this